Business/Repository/Company/CompanyRepository.cs

```diff
@@ -13,8 +13,8 @@ public CompanyRepository(IDocumentQueryService documentQueryService) : base(docu
         public CompanyDto GetCompany()
         {
             return DocumentQueryService.GetDocuments<CMS.DocumentEngine.Types.MedioClinic.Company>()
-                .AddColumns("CompanyName", "Street", "City", "Country",
-            "ZipCode", "PhoneNumber", "Email", "DocumentID")
+                .Columns("CompanyName", "Street", "City", "Country",
+            "ZipCode", "PhoneNumber", "Email", "DocumentID", "NodeSiteId")
```

Comment:
@eugene-cp The reason for using 'AddColumns' instead of 'NodeSiteId' is so that developers do not have to think about adding 'NodeSiteId' to every call they make. Is there any particular reason 'AddColumns' cannot be used? As far as I remember, column selection worked fine when investigating raw SQL queries in SQL profiler.

Reply:
@Enngage NodeSiteId is required in preview mode for rendering FILE fields, it would be great to just add them by default to document queries.  Here's the error on preview mode if NodeSiteId is not available:

![image](https://user-images.githubusercontent.com/22228592/51484407-1736a080-1d61-11e9-89ff-f28d59ade543.png)

As to the comparison in using .AddColumns and .Columns, please see the following:
- using AddColumns - returns 166 fields/Columns
![k12-addcolumns](https://user-images.githubusercontent.com/22228592/51484475-3df4d700-1d61-11e9-9743-5bcceb554d74.PNG)
- using Columns - returns only 6 fields/columns
![k12-columns](https://user-images.githubusercontent.com/22228592/51484486-4a792f80-1d61-11e9-92f9-886a656eab2d.PNG)

Reply:
@Enngage also an alternative, is to just set .Columns("NodeSiteId") in the DocumentQueryService so it is added only once and then use .AddColumns to the repository classes.

Example: 
- DocumentQueryService.cs
```
...
        public DocumentQuery<TDocument> GetDocuments<TDocument>() where TDocument : TreeNode, new()
        {
            var query = DocumentHelper.GetDocuments<TDocument>();

            // Loads the latest version of documents as preview mode is enabled
            if (SiteContext.IsPreviewEnabled)
            {
                query = query
                    .Columns("NodeSiteId")
                    .OnSite(SiteContext.SiteName) // There could be more sites with matching documents
                    .LatestVersion()
                    .Published(false)
                    .Culture(SiteContext.PreviewCulture);
            } else {
                query = query
                    .OnSite(SiteContext.SiteName) // There could be more sites with matching documents
                    .Published()
                    .PublishedVersion()
                    .Culture(SiteContext.CurrentSiteCulture);
            }

            return query;
        }
...
```

- DoctorRepository.cs
```
        private readonly string[] _doctorColumns = 
        {
            // Defines database columns for retrieving data
            // NodeGuid is retrieved automatically
            "NodeID", "NodeAlias", "Bio", "Degree", "EmergencyShift", "FirstName",
            "LastName", "Photo", "Specialty", "DocumentID"
        };

        public IEnumerable<DoctorDto> GetDoctors()
        {
            return DocumentQueryService.GetDocuments<CMS.DocumentEngine.Types.MedioClinic.Doctor>()
                .AddColumns(_doctorColumns)
                .ToList()
                .Select(DoctorDtoSelect);
        }

        public DoctorDto GetDoctor(Guid nodeGuid)
        {
            return DocumentQueryService.GetDocument<CMS.DocumentEngine.Types.MedioClinic.Doctor>(nodeGuid)
                .AddColumns(_doctorColumns)
                .Select(DoctorDtoSelect)
                .FirstOrDefault();
        }
```

Reply:
@eugene-cp Thank you for such a detailed response! I think you are spot on (your latest comment with having the `NodeSiteId` in `DocumentQueryService`. The thing is though that this was exactly what we had before - see [this commit](https://github.com/Kentico/training-kentico-k12/commit/2fb00f2225dc7ba734c024f8e4df3c5933f91b38#diff-c33611460b7149551bdd5013f092b579). 

I was under the impression that @JanLenoch removed the `NodeSiteID` because it was no longer required for preview mode and Kentico added it automatically. If that is not the case (and it doesn't seem to be) I would reintroduce the original implementation and what you already suggested. 

I'm not sure if the `DocumentService` should have `AddColumns` or just 'Columns'. In either case the result should be that there are no `select * from` queries run against db. Do you think you could update your PR and verify this for both preview enabled/disabled?

Reply:
@Enngage revised the code, please see below capture from Glimpse

- Preview Mode
![image](https://user-images.githubusercontent.com/22228592/51538949-35aba300-1e18-11e9-8d0c-54d90d5da126.png)

- Live Mode
![image](https://user-images.githubusercontent.com/22228592/51539168-f29dff80-1e18-11e9-8eb5-5ee9423d622e.png)